src/platform/backends/qemu/qemu_virtual_machine.cpp

```diff
@@ -190,17 +243,29 @@ mp::QemuVirtualMachine::QemuVirtualMachine(const VirtualMachineDescription& desc
 
 mp::QemuVirtualMachine::~QemuVirtualMachine()
 {
-    remove_tap_device(QString::fromStdString(tap_device_name));
-
     update_shutdown_status = false;
-    shutdown();
+    if (state == State::running)
+        suspend();
```

Comment:
Suspend/resume is faster than booting it up again?

Reply:
Yes, by far.  On my machine boot takes ~35-40s, whereas resume from a saved memory snapshot takes ~5-10s.  Also, saving the memory snapshot takes less time than cleanly shutting down the machine too.